### PR TITLE
MFB-851: Add WA EITC (wa)

### PR DESCRIPTION
**Pull request by elliottwaissbluth**
- [x] AI code draft written
 - [x] Coderabbit review
 - [x] Coderabbit comments addressed
 - [x] Claude review
 - [x] Claude comments addressed
 - [ ] Playwright validations

## Context & Motivation

- Fixes [MFB-851](https://linear.app/myfriendben/issue/MFB-851/wa-eitc)

Adds the federal Earned Income Tax Credit (EITC) as a PolicyEngine-based program under the WA white label. EITC is a refundable federal tax credit for workers with low to moderate income.

## Changes Made

- **Research files**: Added spec (`programs/programs/wa/eitc/spec.md`), initial config (`wa_eitc_initial_config.json`), and validation test cases (`wa_eitc.json`)
- **Program class**: Added `WaEitc` to `programs/programs/wa/pe/tax.py` — wraps the federal `Eitc` PE calculator with `WaStateCodeDependency`
- **Calculator registry**: Registered `wa_eitc` in `wa_pe_calculators` and `wa_tax_calculators`
- **Tests**: Added `TestWaEitc` wiring tests in `programs/programs/wa/pe/tests/test_tax.py`
- **Initial config**: Added `wa_tax` program category (icon: `tax_credit`, `tax_category: true`)
- **Validations**: 3 test cases — golden path eligible, income above threshold ineligible, investment income edge case

## Testing

- Migrations to run: None (no model changes)
- Configuration updates needed: Run `import_program_config` with `wa_eitc_initial_config.json`, then `import_validations` with `wa_eitc.json`
- Environment variables/settings to add: None
- Manual testing steps:
  ```bash
  python manage.py import_program_config programs/management/commands/import_program_config_data/data/wa_eitc_initial_config.json --skip-translation
  python manage.py import_validations validations/management/commands/import_validations/data/wa_eitc.json
  # Set value_type and activate:
  python manage.py shell -c "from programs.models import Program; p = Program.objects.get(name_abbreviated='wa_eitc'); p.value_type = 'tax_credit'; p.active = True; p.save()"
  python manage.py validate -w wa -p wa_eitc
  ```

## Deployment

- Post-deployment scripts needed: Run `import_program_config` and `import_validations` commands above (without `--skip-translation`)
- Production config updates: Set `value_type = 'tax_credit'` and `active = True` on the wa_eitc program
- Admin updates needed: None
- Notify team/users of: New EITC program available on WA white label

## Notes for Reviewers

- Known limitations:
  - **Investment income cap not enforced by PE**: The screener maps investment income to `capital_gains`, but PE's `eitc_relevant_investment_income` reads from `net_investment_income` and `net_capital_gains` (which have their own component variables). This means the $11,950 TY2025 investment income cap is not enforced. This is a systemic PE input mapping limitation affecting all EITC implementations (TX, WA, etc.), not specific to this PR.
  - **PE computes actual EITC, not max credit**: The spec describes showing the max credit for the filer's tier, but PE computes the actual phase-in/phase-out amount. Test case expected values match PE's actual output.
  - **wa_wic has no calculator registered**: Pre-existing issue — `wa_wic` is active but has no PE calculator class, causing `KeyError` when running full WA validations. Temporarily deactivated during validation testing.
- Future considerations:
  - Consider adding `value_type` to `import_program_config` simple_config_fields to avoid manual shell command
  - Investment income cap could be enforced via a custom pre-check or by fixing PE input mapping upstream

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added support for Washington Earned Income Tax Credit (EITC) program for tax year 2025, including eligibility criteria, benefit calculations, and program configuration.

* **Documentation**
  * Added comprehensive program specification covering eligibility requirements, benefit values, data gaps, and test scenarios for Washington EITC.

* **Tests**
  * Added validation test cases and integration tests for Washington EITC calculator functionality.

[Review Change Stack](https://app.coderabbit.ai/change-stack/MyFriendBen/benefits-api/pull/1500)

### CODE REVIEW

**Comment by elliottwaissbluth** [comment]
# Code Review: MFB-851

**PR**: [#1500](https://github.com/MyFriendBen/benefits-api/pull/1500) - MFB-851: Add WA EITC (wa)
**Ticket**: [MFB-851](https://linear.app/myfriendben/issue/MFB-851/wa-eitc)
**Author**: Elliott Waissbluth
**Type**: Backend (Django/Python)
**Reviewed**: 2026-05-11

---

## PR Overview

This PR adds the federal Earned Income Tax Credit (EITC) as a PolicyEngine-based program under the WA white label. EITC is a refundable federal tax credit for workers with low to moderate income — it can lower taxes owed and may increase a refund even for workers without children.

The implementation follows the established pattern for PE-backed state tax programs: a thin wrapper class (`WaEitc`) inherits federal EITC inputs from the shared `Eitc` calculator, adds the WA state code dependency so PolicyEngine applies state-specific context, and outputs the federal `eitc` dollar amount. This is the same pattern used by `TxEitc` (Texas) and closely mirrors `WaWftc` (WA Working Families Tax Credit) which already exists in the same file.

### What Problem Does This Solve?

Washington residents using the MFB screener can now be screened for the federal EITC. Previously, EITC screening was only available on the TX white label. Since EITC is the largest federal antipoverty tax credit (~$68.5B annually, ~23.5M filers), adding it to WA's program lineup significantly expands benefit discovery for Washington users.

### How Does It Work?

#### New PolicyEngine Calculator Wrapper

- Adds `WaEitc` class in `programs/programs/wa/pe/tax.py` — a 7-line class that sets `pe_name = "eitc"`, inherits all federal `Eitc.pe_inputs`, adds `WaStateCodeDependency`, and outputs `dependency.tax.Eitc`.
- Registers `wa_eitc` in both `wa_tax_calculators` and `wa_pe_calculators` in `programs/programs/wa/pe/__init__.py`.

#### Program Configuration and Research

- Adds `wa_eitc_initial_config.json` with program metadata (name, description, legal statuses, documents needed, navigator info, application links).
- Adds a comprehensive `spec.md` (485 lines) documenting all 17 eligibility criteria, 20 data gaps, 13 test scenarios, and benefit value tables for TY2025.

#### Validation Test Cases

- Adds `wa_eitc.json` with 3 validation scenarios: golden-path eligible (single parent, 2 children), income-too-high ineligible (childless filer), and investment income edge case (documents PE limitation where investment income cap is not enforced).

#### Wiring Tests

- Adds `TestWaEitc` test class with 10 tests verifying calculator subclass, `pe_name`, registry presence in all three dictionaries, input/output dependencies, and that exactly one dependency is added on top of the federal `Eitc` inputs.

---

## Visual Architecture

### Calculator Inheritance Diagram

```mermaid
graph TB
    Base[PolicyEngineTaxUnitCalulator] --> FedEitc[Eitc<br/>pe_name: eitc]
    Base --> WaWftc[WaWftc<br/>pe_name: wa_working_families_tax_credit]
    Base --> WaEitc[WaEitc<br/>pe_name: eitc]
    Base --> TxEitc[TxEitc<br/>pe_name: eitc]

    FedEitc -.->|pe_inputs inherited| WaEitc
    FedEitc -.->|pe_inputs inherited| WaWftc
    FedEitc -.->|pe_inputs inherited| TxEitc

    WaEitc -->|+WaStateCodeDependency| PE[PolicyEngine API]
    WaWftc -->|+WaStateCodeDependency| PE
    TxEitc -->|+TxStateCodeDependency| PE

    PE -->|eitc dollar amount| EitcOut[dependency.tax.Eitc]
    PE -->|wftc dollar amount| WftcOut[dependency.tax.WaWftc]

    style WaEitc fill:#c8e6c9,stroke:#2e7d32
    style FedEitc fill:#e1f5ff,stroke:#0277bd
    style PE fill:#fff3e0,stroke:#ef6c00
```

### Registry Flow

```mermaid
graph LR
    WaEitc[WaEitc class] -->|registered as 'wa_eitc'| TaxCalcs[wa_tax_calculators]
    TaxCalcs -->|spread into| PeCalcs[wa_pe_calculators]
    PeCalcs -->|spread into| Global[all_tax_unit_calculators]
    Global -->|used by| Engine[PE Calculation Engine]

    style WaEitc fill:#c8e6c9
    style Global fill:#e1f5ff
    style Engine fill:#fff3e0
```

### Data Flow for EITC Calculation

```mermaid
sequenceDiagram
    participant Screen as Screener Input
    participant Calc as WaEitc Calculator
    participant PE as PolicyEngine API
    participant DB as Database

    Screen->>Calc: Screen with household data
    Note over Calc: Gathers pe_inputs:<br/>- AgeDependency<br/>- TaxUnitSpouseDependency<br/>- TaxUnitDependentDependency<br/>- IRS gross income breakdown<br/>- WaStateCodeDependency (state=WA)
    Calc->>PE: POST simulation request
    PE->>PE: Compute federal EITC<br/>(phase-in, phase-out,<br/>QC rules, age rules)
    PE-->>Calc: eitc dollar amount
    Calc-->>DB: Store eligibility result
    DB-->>Screen: Return EITC eligibility + value
```

---

## Code Review

### 1. WaEitc Calculator Class — Clean Pattern Match

**`programs/programs/wa/pe/tax.py`, lines 6–12 (PR branch)**

The new `WaEitc` class is structurally identical to `TxEitc` in `programs/programs/tx/pe/tax.py`. It correctly:
- Inherits from `PolicyEngineTaxUnitCalulator` (note: the base class has a typo "Calulator" — this is pre-existing, not introduced here)
- Spreads `Eitc.pe_inputs` and adds `WaStateCodeDependency`
- Outputs `dependency.tax.Eitc` (the federal EITC value, not a state-specific wrapper)

```diff
+class WaEitc(PolicyEngineTaxUnitCalulator):
+    pe_name = "eitc"
+    pe_inputs = [
+        *Eitc.pe_inputs,
+        dependency.household.WaStateCodeDependency,
+    ]
+    pe_outputs = [dependency.tax.Eitc]
```

No issues. This follows the established pattern exactly.

### 2. Missing Docstring on WaEitc (Minor — Consider Adding)

**`programs/programs/wa/pe/tax.py`, lines 6–12**

`WaWftc` in the same file has a detailed docstring explaining what it does, why it exists, and what screener gaps it doesn't block on. `WaEitc` has none. While the CLAUDE.md says "default to writing no comments," a brief docstring would maintain consistency within this file and help future developers understand the distinction between EITC (federal credit, computed by PE) vs WFTC (state credit, also computed by PE).

```diff
 class WaEitc(PolicyEngineTaxUnitCalulator):
+    """Federal Earned Income Tax Credit — WA white label wrapper."""
     pe_name = "eitc"
     pe_inputs = [
         *Eitc.pe_inputs,
```

This is a style nit, not a blocking concern. The `WaWftc` docstring sets a local precedent in this file, so a one-liner maintains consistency.

### 3. Calculator Registration — Correct

**`programs/programs/wa/pe/__init__.py`, line 16**

```diff
 wa_tax_calculators = {
+    "wa_eitc": tax.WaEitc,
     "wa_wftc": tax.WaWftc,
 }
```

Alphabetical ordering is maintained. The `wa_pe_calculators` dict on line 19 already spreads `**wa_tax_calculators`, so `wa_eitc` automatically flows into both `wa_pe_calculators` and (via the registry) `all_tax_unit_calculators`. No separate registration needed.

### 4. Test Import Update — Correct

**`programs/programs/wa/pe/tests/test_tax.py`, line 27**

```diff
-from programs.programs.wa.pe.tax import WaWftc
+from programs.programs.wa.pe.tax import WaEitc, WaWftc
```

Clean addition. Tests are thorough — 10 test methods covering subclass identity, pe_name, all 3 registries, input/output dependencies, and dependency count. The pattern mirrors `TestWaWftc` exactly.

### 5. Validation Test Case 3 — Investment Income Edge Case (Noteworthy)

**`validations/management/commands/import_validations/data/wa_eitc.json`, test case 3**

The third validation case is the most interesting: it tests a household with investment income of $1,000/month ($12,000/year), which is **above** the TY2025 $11,950 investment income limit. Per IRS rules, this household should be *ineligible*. However, the test expects `eligible: true` with `value: 1347`.

This is intentionally documenting a **known PE limitation**: the screener maps investment income to `capital_gains`, but PE's `eitc_relevant_investment_income` reads from `net_investment_income` and `net_capital_gains` (which have their own component variables). The $11,950 cap is therefore not enforced by PE.

The PR description clearly documents this limitation and notes it affects all EITC implementations (TX, WA, etc.), not just this PR. The validation note also explains this. This is the correct approach — documenting the limitation in the test rather than pretending the cap works.

**No action needed**, but this is worth noting for future work on PE input mapping.

### 6. Initial Config — `value_type` Field

**`programs/management/commands/import_program_config_data/data/wa_eitc_initial_config.json`, line 38**

The config includes `"value_type": "tax_credit"`, which is correct. The PR description mentions needing a manual shell command to set `value_type` post-import because it's not in `import_program_config`'s `simple_config_fields`. The PR notes also mention this as a future improvement.

However, looking at the config file, `value_type` IS included in the JSON. If `import_program_config` doesn't handle it, the field will be silently ignored during import. This is consistent with the PR's explicit instruction to run:

```python
p.value_type = 'tax_credit'; p.active = True; p.save()
```

No issue with the PR itself — the deployment instructions are clear. But the `value_type` being in the JSON but requiring a manual shell command is a minor footgun for future programs.

### 7. Missing Newline at End of Config File

**`programs/management/commands/import_program_config_data/data/wa_eitc_initial_config.json`, line 99**

```diff
     }
   ]
-}
\ No newline at end of file
+}
```

The file is missing a trailing newline. This is a minor nit that may trigger linter warnings and produces a noisy diff on future edits.

---

## Security Analysis

- **Input Validation**: PASS — No user input handling in this PR. All data flows through existing validated paths (screener → PE API).
- **SQL Injection**: PASS — No raw SQL. All data access goes through Django ORM via existing patterns.
- **XSS Prevention**: PASS — No frontend changes. Backend only.
- **Authentication/Authorization**: PASS — No new endpoints. Program eligibility is computed within existing authenticated flows.
- **Sensitive Data**: PASS — No PII handling. No secrets in code. Config file contains only program metadata and public URLs.
- **CSRF Protection**: PASS — No new views or endpoints.
- **Dependencies**: PASS — No new dependencies added.

**Critical Issues:** None

**Recommendations:** None — this is a low-risk, pattern-following change with no new attack surface.

---

## Impact Analysis

### What Changed

| File | Change |
|------|--------|
| `programs/programs/wa/pe/tax.py` | Added `WaEitc` class (7 lines) |
| `programs/programs/wa/pe/__init__.py` | Registered `wa_eitc` in `wa_tax_calculators` |
| `programs/programs/wa/pe/tests/test_tax.py` | Added `TestWaEitc` (10 tests, 48 lines) |
| `programs/programs/wa/eitc/spec.md` | New research spec (485 lines) |
| `wa_eitc_initial_config.json` | New program config (93 lines) |
| `wa_eitc.json` | New validation cases (189 lines, 3 scenarios) |

### Potential Impact

**Direct Dependencies:**
- `wa_pe_calculators` dict — `wa_eitc` is now included. Any code iterating this dict will process one additional program.
- `all_tax_unit_calculators` global registry — same.
- No database schema changes (no migrations).
- No API contract changes.

**Indirect Dependencies:**
- WA white label program list — after `import_program_config` and activation, `wa_eitc` will appear in screening results.
- Full WA validation suite — running `python manage.py validate -w wa` will now include `wa_eitc`. PR notes mention `wa_wic` has a pre-existing `KeyError` when running full WA validations (no calculator registered).

**Breaking Changes:**
- None. This is purely additive — a new calculator class, a new entry in existing dicts. No existing behavior changes.

### Recommended Additional Testing

- Run `python manage.py validate -w wa -p wa_eitc` to confirm all 3 validation cases pass against PolicyEngine
- Manually test Scenarios 1–13 from `spec.md` via the WA screener UI after activation
- Verify the `wa_wic` KeyError is independent of this PR (pre-existing)

---

## Business Logic Correctness

**Spec compared**: `programs/programs/wa/eitc/spec.md`

This PR uses a PolicyEngine-based calculator (not a custom calculator), so the eligibility logic is delegated entirely to PE. The `WaEitc` class does not implement any of the 17 eligibility criteria from the spec — PE handles income thresholds, qualifying child rules, age requirements, and phase-in/phase-out calculations internally.

**Correctly delegated to PE:**
- Criterion 1 (earned income requirement) — PE evaluates via IRS gross income inputs
- Criterion 2 (AGI below income thresholds by filing status/child count) — PE computes phase-in/phase-out
- Criterion 3 (earned income below same thresholds) — PE evaluates both AGI and earned income
- Criterion 6 (U.S. citizen/resident alien) — proxied by `legal_status_required` in config
- Criterion 8 (age 25-64 for childless filers) — PE evaluates via `AgeDependency`
- Criteria 9-11 (qualifying child tests) — PE evaluates via `TaxUnitDependentDependency`

**Known PE limitations documented:**
- Criterion 4 (investment income cap $11,950) — PE does NOT enforce this due to input mapping limitation. Documented in PR description, validation test case 3, and spec. This is a systemic issue affecting all EITC implementations.
- Criterion 5 (MFS filing status) — screener cannot detect MFS; assumes MFJ for married couples. Documented in spec.

**Discrepancies found:**
- **Spec says "screener shows maximum credit"** (Benefit Value section), but **PE computes actual phase-in/phase-out amount**. The PR description explicitly notes this: "PE computes actual EITC, not max credit." Validation test case 1 expects `value: 4993` for a household with $33,600 income and 2 children — the max credit for 2 children is $7,152, so the $4,993 value is the PE-calculated phase-in amount. This is actually more accurate than showing the max credit, but it contradicts the spec's stated methodology. The spec should be updated to reflect the actual PE behavior, or the PR description should note the spec divergence.

**Spec rules not covered by implementation (by design):**
- Criteria 5, 7, 12, 13, 14 — data gaps (assumptions documented in spec and description)
- Criteria 15, 16, 17 — proxied by config-level settings and screener assumptions
- Data gaps 18, 19, 20 — acknowledged, no screener fields available

---

## Migration Safety

N/A — no migrations in this PR.

---

## Checklist

- [x] Code follows project conventions (Django/React patterns)
- [x] Tests are comprehensive and passing (10 wiring tests)
- [x] No security vulnerabilities introduced
- [x] Breaking changes documented and migrations provided (none needed)
- [ ] N/A — TypeScript types (no frontend changes)
- [x] Database queries are optimized (no new queries)
- [x] Error handling is robust (delegates to existing PE flow)
- [ ] N/A — Accessibility (no frontend changes)
- [ ] Documentation: spec methodology says "max credit" but PE returns actual phase-in/phase-out amount
- [x] Business logic matches program spec (PE delegation, known limitations documented)
- [x] Migration safety verified (no migrations)

---

## Summary

**Recommendation**: APPROVE

**Key Takeaways:**
1. **Clean, pattern-following implementation** — `WaEitc` is structurally identical to `TxEitc`, follows established PE wrapper conventions, and adds comprehensive wiring tests. The 826 additions are almost entirely research artifacts (spec, config, validations) with only ~10 lines of production code.
2. **Known PE limitation well-documented** — The investment income cap ($11,950) is not enforced by PE due to input mapping. This is a systemic issue affecting all EITC implementations and is transparently documented in the PR description, spec, and validation test case.
3. **Minor spec/implementation divergence** — The spec states the screener shows the "maximum credit for their tier," but PE actually computes the precise phase-in/phase-out amount (which is more accurate). Suggest updating the spec's Benefit Value methodology section to reflect PE's actual behavior.

**Next Steps:**
- Merge after confirming `python manage.py validate -w wa -p wa_eitc` passes
- Run `import_program_config` and `import_validations` post-merge
- Set `value_type = 'tax_credit'` and `active = True` on the wa_eitc program in production
- Consider updating spec.md Benefit Value section to clarify PE computes actual (not max) credit

---

*Review generated by Claude Code - Claude Opus 4.6*
*Linear Ticket: [MFB-851](https://linear.app/myfriendben/issue/MFB-851/wa-eitc)*
*GitHub PR: [#1500](https://github.com/MyFriendBen/benefits-api/pull/1500)*

File: programs/management/commands/import_program_config_data/data/wa_eitc_initial_config.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      "refugee",
+      "otherWithWorkPermission"
+    ],
+    "name": "Earned Income Tax Credit (EITC)",
+    "description": "The Earned Income Tax Credit (EITC) is a federal tax credit for workers with low to moderate income. It can lower the taxes you owe and may increase your refund, even if you do not have children or are self-employed.\n\nYou claim the EITC by filing a federal tax return. The amount depends on your income, filing status, and how many qualifying children you claim. This screener shows an estimated maximum credit, so your actual refund may be lower.\n\nSome EITC rules cannot be checked here. You, your spouse, and any qualifying children must have Social Security numbers that are valid for work. You also may need to meet rules about living in the United States for more than half the year, not being claimed as someone else's dependent, not filing Form 2555, and filing jointly if married unless a separated-spouse rule applies.\n\nFile a federal tax return to claim the EITC, even if you owe no tax. You can file for free through IRS Free File or get free help from a VITA tax site. Washington's Working Families Tax Credit is separate and may have different rules.",
```
[the remark this answers is not in the document]
Comment (elliottwaissbluth) [reply programs/management/commands/import_program_config_data/data/wa_eitc_initial_config.json:23]: this is fine

File: validations/management/commands/import_validations/data/wa_eitc.json [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+          }
+        }
+      ],
+      "expenses": []
+    },
+    "expected_results": {
+      "program_name": "wa_eitc",
+      "eligible": true,
+      "value": 1347
+    }
```
[the remark this answers is not in the document]
Comment (elliottwaissbluth) [reply validations/management/commands/import_validations/data/wa_eitc.json:187]: The validation suite (python manage.py validate) is an integration test against PolicyEngine's actual output — it asserts what the calculator does produce, not what the law says should happen. If you set eligible: false but PE returns eligible: true, the test fails on every single run. That means:

You can't use validate as a CI/regression gate — it will always be red
Real regressions get masked by the permanent failure — the exact opposite of what the reviewer wants
The harness doesn't have a skip/gate mechanism — there's no way to mark a case as "expected wrong, skip until fixed"
The PR's current approach is correct: assert PE's actual output (eligible: true, value: 1347), document the statutory divergence clearly in the test notes, and track the PE input mapping fix as a separate concern. The test is providing regression value right now — if PE ever fixes their eitc_relevant_investment_income mapping, this test will break and signal that the expected value changed.

The reviewer's underlying concern — don't silently accept wrong answers — is legitimate. But the right remedy is a tracked issue for the PE input mapping fix (which the PR description already calls out as a future consideration), not a validation case that fails on every run.
